cvat/apps/engine/serializers.py

```diff
@@ -1199,6 +1190,47 @@ def _iterate_chunk_frames():
 
         return instance
 
+    @staticmethod
+    def _write_updated_static_chunk(
+        db_task_id: int,
+        db_segment_id: int,
+        chunk_id: int,
+        chunk_frames: list[int],
+        quality: FrameQuality,
+        frame_path_map: dict[int, str],
+        segment_frame_map: dict[int,int]
+    ):
+        from cvat.apps.engine.frame_provider import prepare_chunk
+
+        db_task = models.Task.objects.get(pk=db_task_id)
+        task_frame_provider = TaskFrameProvider(db_task)
+        db_data = db_task.data
+
+        def _iterate_chunk_frames():
+            for chunk_frame in chunk_frames:
+                db_frame_path = frame_path_map[chunk_frame]
+                chunk_real_frame = segment_frame_map.get(chunk_frame, chunk_frame)
+                yield (
+                    task_frame_provider.get_frame(
+                        chunk_real_frame, quality=quality
+                    ).data,
+                    os.path.basename(db_frame_path),
+                    chunk_frame,
+                )
+
+        with closing(_iterate_chunk_frames()) as frame_iter:
+            chunk, _ = prepare_chunk(
+                frame_iter, quality=quality, db_task=db_task, dump_unchanged=True,
+            )
+
+            get_chunk_path = {
+                FrameQuality.COMPRESSED: db_data.get_compressed_segment_chunk_path,
+                FrameQuality.ORIGINAL: db_data.get_original_segment_chunk_path,
+            }[quality]
+
+            with open(get_chunk_path(chunk_id, db_segment_id), 'wb') as f:
+                f.write(chunk.getvalue())
```

Comment:
As we don't obtain a lock for a task and don't prefetch related models (segments, images, etc.), the task can updated during the chunk preparation. In this case there can be any mix of the previous and updated frames in the chunk. I think, it makes sense to remember the chunk updated date before chunk preparation and then check if it's still the same right before writing the updated chunk to the disk. If not, then discard, and, maybe, retry.